packages/salesforce-adapter/src/filters/utils.ts

```diff
@@ -455,8 +455,8 @@ export const instanceInternalId = (instance: InstanceElement): string =>
 
 export const hasApiName = (elem: Element): boolean => apiName(elem) !== undefined
 
-export const extractFlatCustomObjectFields = async (elem: Element): Promise<Element[]> =>
-  (await isCustomObject(elem)) && isObjectType(elem) ? [elem, ...Object.values(elem.fields)] : [elem]
+export const extractFlatCustomObjectFields = (elem: Element): Element[] =>
+  isCustomObjectSync(elem) && isObjectType(elem) ? [elem, ...Object.values(elem.fields)] : [elem]
```

Comment:
```suggestion
  [elem].concat(isCustomObjectSync(elem) ? Object.values(elem.fields) as Element[] : [])
```
- Now that you use `isCustomObjectSync` there's no need to call `isObjectType`.
- Avoid using spread operator due to possible OOM

Reply:
Done.